### PR TITLE
Make untracked return value & effect handle teardown

### PR DESCRIPTION
**Pull request by dy**
Hi @WebReflection

I know you mentioned there's no fancy features in this library, but I took liberty to propose 2 updates.

1. `untracked` function seemingly must [return a value](https://github.com/preactjs/signals?tab=readme-ov-file#untrackedfn) - I missed that out making #5 
2. `effect` callback can return a _teardown_ (like `useEffect`) - not directly documented in preact, but very handy feature, low-hanging fruit. It's minor breakage of existing API, but in favor of more standardization.

Please let me know if you find value in these changes.

### CODE REVIEW

File: esm/index.js [hunk 1]
```diff
@@ -51,8 +51,8 @@ const create = block => {
  * @template T
  * @type {<T>(fn: (v?: T) => T | undefined, value?: T) => () => void}
  */
-export const effect = (fn, value) => {
-  const fx = create(() => { value = fn(value) });
+export const effect = (fn) => {
+  let teardown, fx = create(() => { teardown?.call?.(); teardown = fn() });
```
Comment (WebReflection) [inline esm/index.js:55]: where is this returned?

Comment (WebReflection) [reply esm/index.js:55]: I can't track `teardown` lifecycle, sorry ...

Comment (dy) [reply esm/index.js:55]: It's called in-between effect calls:

```js
effect(() => {
  // main effect code, eg.
  let id = setInterval(() => count++, 1000)
  return () => {
    // calls before next effect call
    clearTimeout(id)
  }
})
```

But you're right, I forgot that we should also run teardown when we dispose effect.

Comment (dy) [reply esm/index.js:55]: @WebReflection please have a look

**Comment by WebReflection** [comment]
done?

**Comment by dy** [comment]
Yep, looks perfect, thanks @WebReflection

**Pull request merged**